### ISSUE TITLE
Firefox can sometimes get unresponsive for some seconds when closing "a lot" (50+) of Temporary Containers at once

### ISSUE THREAD

**Issue by stoically** [stoically/firefox-add-on-temporary-containers#47]
Should be solvable by not deleting all containers at once but in intervals

Although this only happened to me once. I can't reproduce it with just opening 50 tabs without any content (besides about:home). So I'm not exactly sure whether Temporary Containers where actually the cause - but I guess it doesn't hurt to delete in batches anyway.

**Comment by stoically** [comment]
Just published version 0.36 with the optimization

**Comment by r-a-y** [comment]
I've noticed this myself.

CPU hovers around 25-35% and takes awhile to get back down when closing a number of tabs at once in Automatic Mode. My guess is that deleting a container takes up most of the cycles. To confirm, I just tried to delete one container manually with the Multi-Account Containers addon and I notice the same CPU usage.

Not much we can really do here except hoping MAC gets better optimization. There are already a number of issues in the MAC repo regarding high CPU usage -  https://github.com/mozilla/multi-account-containers/issues?q=cpu.
